source/core_json.c

```diff
@@ -1346,39 +1582,18 @@ JSONStatus_t JSON_Search( char * buf,
     }
     else
     {
-        while( i < queryKeyLength )
-        {
-            start = i;
-
-            while( ( i < queryKeyLength ) && ( queryKey[ i ] != separator ) )
-            {
-                i++;
-            }
-
-            keyLength = i - start;
-
-            /* catch an empty key subpart or a trailing separator */
-            if( ( keyLength == 0U ) || ( i == ( queryKeyLength - 1U ) ) )
-            {
-                ret = JSONBadParameter;
-                break;
-            }
-
-            i++;
-
-            ret = search( p, tmp, &queryKey[ start ], keyLength, &p, &tmp );
-
-            if( ret != JSONSuccess )
-            {
-                break;
-            }
-        }
+        ret = multiSearch( buf, max, queryKey, queryKeyLength, outValue, outValueLength );
     }
 
     if( ret == JSONSuccess )
     {
-        *outValue = p;
-        *outValueLength = tmp;
+        /* String values and collections include their surrounding
+         * demarcation.  If the value is a string, strip the quotes. */
+        if( *outValue[ 0 ] == '"' )
+        {
+            ( *outValue )++;
```

Comment:
Shouldn't this be an increment of the pointer address value instead of the pointed _to_ value?
```suggestion
            outValue++;
```

Reply:
outValue is a char **.  If this were wrong, the unit tests would catch it.